outwatch/src/main/scala/outwatch/util/Store.scala

```diff
@@ -1,64 +1,29 @@
 package outwatch.util
 
 import cats.effect.Sync
-import cats.implicits._
-import monix.eval.Task
 import monix.execution.Scheduler
-import monix.reactive.{ObservableLike, Observable}
+import monix.eval.Task
 import monix.reactive.subjects.PublishSubject
 import org.scalajs.dom
 import outwatch._
 
 object Store {
 
-  /**
-   * A Function that applies an Action onto the Stores current state.
-   * @tparam A The Action Type
-   * @tparam M The Model Type
+  /* Partial application trick, "kinda-curried type parameters"
+   * https://typelevel.org/cats/guidelines.html
    */
-  type Reducer[A, M] = (M, A) => (M, Observable[A])
-
-  object Reducer {
-    /**
-     * Creates a Reducer which yields a new State, as-well as an Observable of Effects
-     * Effects are Actions which will be executed after the Action that caused them to occur.
-     * This is accomplished by subscribing to the Effects Observable within the stores scan loop.
-     *
-     * CAUTION: There is currently a bug which causes the Effect-States to emit,
-     * before the State of the action that caused the effects is emitted.
-     * However, this only effects immediate emissions of the Effects Observable, delayed emissions should be fine.
-     * @param f The Reducing Function returning the (Model, Effects) tuple.
-     */
-    def withEffects[A, M](f: (M, A) => (M, Observable[A])): Reducer[A, M] = f
-
-    /**
-     * Creates a reducer which just transforms the state, without additional effects.
-     */
-    def apply[A, M](f: (M, A) => M): Reducer[A, M] = (s: M, a: A) => f(s, a) -> Observable.empty
-
-    /**
-     * Creates a Reducer with an optional IO effect.
-     */
-    def withOptionalEffects[F[_]: ObservableLike, A, M](f: (M, A) => (M, Option[F[A]])): Reducer[A, M] = { (s: M, a: A) =>
-      f(s, a).map(_.fold[Observable[A]](Observable.empty)(Observable.from))
-    }
+  final class CreatePartiallyApplied[F[_]](val dummy: Boolean = false) extends AnyVal {
```

Comment:
Alternatively, instead of a dummy val, we could try `@inline class` and `@inline def` and check whether the scala-js compiler optimizes it properly.

Reply:
I have tested this in fastOpt and inlining does reliably remove the overhead of this partial application trick (see https://github.com/OutWatch/outwatch/pull/343/commits/1d4df805bb92cb6eb298f1642d299df56b64fa99).

For reference, the code generated for `Handler.create[IO, Int](0)` is:
```javascript
  $m_Loutwatch_Handler$().create__O__Lcats_effect_Sync__O(0, $m_Lcats_effect_IO$().ioEffect$2);
```

The generated code for `Handler.create[IO](0)` as it is now with AnyVal:
 ```javascript
$m_Loutwatch_Handler$CreatePartiallyApplied$().apply$extension__Z__O__Lcats_effect_Sync__O(false, 0, $m_Lcats_effect_IO$().ioEffect$2);

// this $m_Loutwatch_Handler$CreatePartiallyApplied$-method somehow constructs a singleton for CreatePartiallyApplied
var $n_Loutwatch_Handler$CreatePartiallyApplied$ = (void 0);
function $m_Loutwatch_Handler$CreatePartiallyApplied$() {
  if ((!$n_Loutwatch_Handler$CreatePartiallyApplied$)) {
    $n_Loutwatch_Handler$CreatePartiallyApplied$ = new $c_Loutwatch_Handler$CreatePartiallyApplied$().init___()
  };
  return $n_Loutwatch_Handler$CreatePartiallyApplied$
}
```

The generated code for `Handler.create[IO](0)` with `@inline` annotations:
 ```javascript
  var F = $m_Lcats_effect_IO$().ioEffect$2;
  $m_Loutwatch_Handler$().create__O__Lcats_effect_Sync__O(0, F);
```

The last snippet generates an extra variable for the implicit `Sync` instance, but nothing else.

Reply:
So, I think for this PR it is good as is. We will work on some part in another PR.